spark/src/main/scala/geotrellis/spark/io/accumulo/AccumuloLayerWriter.scala

```diff
@@ -41,17 +40,51 @@ class AccumuloLayerWriter[K: Boundable: JsonFormat: ClassTag, V: ClassTag, Conta
       case e: Exception => throw new LayerWriteError(id).initCause(e)
     }
   }
+
+  def update(id: LayerId, rdd: RDD[(K, V)]) = {
+    try {
+      if (!attributeStore.layerExists(id)) throw new LayerNotExistsError(id)
+      implicit val mdFormat = cons.metaDataFormat
+      val header =
+        AccumuloLayerHeader(
+          keyClass = classTag[K].toString(),
+          valueClass = classTag[V].toString(),
+          tileTable = table
+        )
+
+      val (existingHeader, existingMetaData, existingKeyBounds, existingKeyIndex, existingSchema) =
+        attributeStore.readLayerAttributes[AccumuloLayerHeader, cons.MetaDataType, KeyBounds[K], KeyIndex[K], Schema](id)
+
+      if (existingHeader != header) throw new HeaderMatchError(id, existingHeader, header)
+
+      val boundable = implicitly[Boundable[K]]
+      val keyBounds = boundable.getKeyBounds(rdd)
+
+      if (!boundable.includes(keyBounds.minKey, existingKeyBounds) || !boundable.includes(keyBounds.maxKey, existingKeyBounds))
+        throw new OutOfKeyBoundsError(id)
+
+      val getRowId = (key: K) => index2RowId(existingKeyIndex.toIndex(key))
+
+      rddWriter.write(rdd, table, columnFamily(id), getRowId, oneToOne = false)
+    } catch {
+      case e: HeaderMatchError[_] => throw e.initCause(e)
+      case e: OutOfKeyBoundsError => throw e.initCause(e)
+      case e: LayerNotExistsError => throw e.initCause(e)
+      case e: Exception => throw new LayerUpdateError(id).initCause(e)
```

Comment:
Even here, I don't usually see the value of catching errors you'll throw, it generally obfuscates the stack trace for me.